### PR TITLE
Michaelc/ora 1130 set tax on workers to be a function of average rewards and

### PR DESCRIPTION
**Pull request by RedBird96**
add tax for only worker reward
[ORA-1130]([ORA-1130 : Set tax on workers to be a function of average rewards and parameters](https://linear.app/upshot/issue/ORA-1130/set-tax-on-workers-to-be-a-function-of-average-rewards-and-parameters))

### CODE REVIEW

**Comment by dmikey** [comment]
so many unrelated commits, makes me sad.

**Comment by RedBird96** [comment]
> so many unrelated commits, makes me sad.

Please review only last commit...

File: x/emissions/proto/emissions/v1/types.proto [hunk 1]
```diff
@@ -269,6 +271,13 @@ message Scores {
   repeated Score scores = 1;
 }
 
+/// AVERAGE REWARD
+
+message AvgReward {
```
Comment (kpeluso) [inline x/emissions/proto/emissions/v1/types.proto:276]: should be consistent -- you say "AverageReward" everywhere but here, where you use shorthand

File: x/emissions/types/keys.go [hunk 2]
```diff
@@ -49,7 +49,8 @@ var (
 	InferenceScoresKey             = collections.NewPrefix(38)
 	ForecastScoresKey              = collections.NewPrefix(39)
 	ReputerListeningCoefficientKey = collections.NewPrefix(40)
-	FeeRevenueEpochKey             = collections.NewPrefix(41)
-	TopicFeeRevenueKey             = collections.NewPrefix(42)
-	PreviousTopicWeightKey         = collections.NewPrefix(43)
+	AverageRewardKey               = collections.NewPrefix(41)
```
Comment (kpeluso) [inline x/emissions/types/keys.go:52]: There are a few actors for which an avg reward can be calculated. this is for workers tho, so should be specified

**Review by kpeluso — CHANGES_REQUESTED** [review]
this is missing 1 key thing per the ticket:
> These funds should be sent to the ecosystem treasury, where all the inference request fees are already paid today.

**Review by kpeluso — APPROVED** [review]
nvm actually... because you're subtracting the tax here, then the rewards will never be sent from the ecosystem bucket => it's equivalent to doing what I said above, just simpler!

great!

File: x/emissions/proto/emissions/v1/types.proto (hunk 1 above)
Comment (kpeluso) [reply x/emissions/proto/emissions/v1/types.proto]: fixed this myself

File: x/emissions/types/keys.go (hunk 2 above)
Comment (kpeluso) [reply x/emissions/types/keys.go]: fixed this myself

**Pull request merged**